src/mono/mono/metadata/native-library.c

```diff
@@ -1050,11 +1050,11 @@ lookup_pinvoke_call_impl (MonoMethod *method, MonoLookupPInvokeStatus *status_ou
 
 	if (!module) {
 		mono_trace (G_LOG_LEVEL_WARNING, MONO_TRACE_DLLIMPORT,
-				"DllImport unable to load library '%s'.",
-				new_scope);
+				"DllImport unable to load library '%s' (originally '%s').",
+				new_scope, orig_scope);
```

Comment:
nit: check whether `new_scope` and `orig_scope` differ so we don't get duplicate library name in the normal case where a library is not redirected.